### PR TITLE
feat: Add has expired column in the api keys table

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Add API key has expired column in the API keys table
- Related with this https://github.com/Flagsmith/flagsmith/pull/3042#pullrequestreview-1885058591

## How did you test this code?

- Go to API keys

### CODE REVIEW

**Comment by matthewelwell** [comment]
Is there anything we can do to make this look a bit nicer? 

image

Perhaps expired isn't the correct data point here, maybe we should have an 'active' column which just includes a tick / cross and add a tooltip explaining if it's because it has expired?

**Comment by novakzaballa** [comment]
> Is there anything we can do to make this look a bit nicer?
> 
> image
> Perhaps expired isn't the correct data point here, maybe we should have an 'active' column which just includes a tick / cross and add a tooltip explaining if it's because it has expired?

Yes, those are good ideas, I'll make those changes.

**Comment by novakzaballa** [comment]
Screen Shot 2024-02-27 at 16 46 20
Hey @matthewelwell I already made the changes, it looks like the image above.

**Comment by matthewelwell** [comment]
image

Can we align things / make things consistent. The active badge should probably be green, right?

**Comment by novakzaballa** [comment]
> image
> Can we align things / make things consistent. The active badge should probably be green, right?

Screen Shot 2024-02-27 at 18 01 20

Done

File: frontend/web/components/AdminAPIKeys.js [hunk 1]
```diff
@@ -481,6 +485,25 @@ export default class AdminAPIKeys extends PureComponent {
                   <Flex className='table-column fs-small lh-sm'>
                     <Switch checked={v.is_admin} disabled={true} />
                   </Flex>
+                  <Flex className='table-column fs-small lh-sm'>
+                    {v.has_expired ? (
+                      <div className='ml-1'>
+                        <Tooltip title={<Icon name='close-circle' />}>
+                          {
+                            'This API key has expired due to reaching its expiration date'
```
Comment (matthewelwell) [inline frontend/web/components/AdminAPIKeys.js]:
```suggestion
                            'This API key has expired'
```

**Review by matthewelwell — APPROVED** [review]
Approved with one minor suggestion on the wording.

File: frontend/web/components/AdminAPIKeys.js (hunk 1 above)
Comment (novakzaballa) [reply frontend/web/components/AdminAPIKeys.js]: Changed

**Pull request merged**